src/instance/minecraft/chat-manager.ts

```diff
@@ -123,4 +129,35 @@ export default class ChatManager extends EventHandler<MinecraftInstance> {
       message
     })
   }
+
+  proceedFiltering(author: string, content: string): string {
+    let filteredMessage: string
+    try {
+      filteredMessage = this.clientInstance.app.profanityFilter.clean(content)
+    } catch {
+      /*
+        profanity package has bug.
+        will throw error if given one special character.
+        example: clean("?")
+        message is clear if thrown
+      */
+      filteredMessage = content
+    }
+
+    if (content !== filteredMessage) {
+      this.clientInstance.app.emit('event', {
+        localEvent: true,
+        instanceType: InstanceType.DISCORD,
+        username: author,
+        message: `${escapeDiscord(author)}:\n\n**Profanity warning, this message has been edited:**\n${escapeDiscord(filteredMessage)}`,
```

Comment:
You can't use discord markup language in the `message` either since the event can be received from anywhere.